sdk/changeanalysis/azure-resourcemanager-changeanalysis/src/test/java/com/azure/resourcemanager/changeanalysis/AzureChangeAnalysisManagerTest.java

```diff
@@ -0,0 +1,81 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.resourcemanager.changeanalysis;
+
+import com.azure.core.credential.TokenCredential;
+import com.azure.core.http.policy.HttpLogDetailLevel;
+import com.azure.core.http.policy.HttpLogOptions;
+import com.azure.core.management.AzureEnvironment;
+import com.azure.core.management.Region;
+import com.azure.core.management.profile.AzureProfile;
+import com.azure.core.test.TestBase;
+import com.azure.core.test.annotation.DoNotRecord;
+import com.azure.core.util.Configuration;
+import com.azure.core.util.CoreUtils;
+import com.azure.identity.DefaultAzureCredentialBuilder;
+import com.azure.resourcemanager.resources.ResourceManager;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.Test;
+
+import java.time.OffsetDateTime;
+import java.util.Random;
+
+public class AzureChangeAnalysisManagerTest extends TestBase {
+
+    private static final Random RANDOM = new Random();
+    private static final Region REGION = Region.US_WEST2;
+    private String resourceGroupName = "rg" + randomPadding();
+    private AzureChangeAnalysisManager azureChangeAnalysisManager;
+    private ResourceManager resourceManager;
+    private boolean testEnv;
+
+    @Override
+    public void beforeTest() {
+        final TokenCredential credential = new DefaultAzureCredentialBuilder().build();
+        final AzureProfile profile = new AzureProfile(AzureEnvironment.AZURE);
+
+        azureChangeAnalysisManager = AzureChangeAnalysisManager
+            .configure()
+            .withLogOptions(new HttpLogOptions().setLogLevel(HttpLogDetailLevel.BASIC))
+            .authenticate(credential, profile);
+
+        resourceManager = ResourceManager
+            .configure()
+            .withLogOptions(new HttpLogOptions().setLogLevel(HttpLogDetailLevel.BASIC))
+            .authenticate(credential, profile)
+            .withDefaultSubscription();
+
+        // use AZURE_RESOURCE_GROUP_NAME if run in LIVE CI
+        String testResourceGroup = Configuration.getGlobalConfiguration().get("AZURE_RESOURCE_GROUP_NAME");
+        testEnv = !CoreUtils.isNullOrEmpty(testResourceGroup);
+        if (testEnv) {
+            resourceGroupName = testResourceGroup;
+        } else {
+            resourceManager.resourceGroups()
+                .define(resourceGroupName)
+                .withRegion(REGION)
+                .create();
+        }
+    }
+
+    @Override
+    protected void afterTest() {
+        if (!testEnv) {
+            resourceManager.resourceGroups().beginDeleteByName(resourceGroupName);
+        }
+    }
+
+    @Test
+    @DoNotRecord(skipInPlayback = true)
+    public void test() {
+        // @embedmeStart
+        Assertions.assertTrue(azureChangeAnalysisManager.changes()
+            .list(OffsetDateTime.now().minusDays(1), OffsetDateTime.now().plusDays(1)).stream().count() > 0);
```

Comment:
Can we use larger time range (e.g. 7 days or 30 days)?

This test depends on env in that subscription. 1 day may be too short to guarantee that there were a change.

Reply:
Fixed in the new version.

Reply:
I didn't see how changing from "1 day in future" to "7 day in future" would help. I don't think the RP can forecast the future changes.

Reply:
The time range has been fixed to within the past two weeks in the new version.